### DISCUSSION TITLE
Nested distinct possible?

### DISCUSSION THREAD

**Discussion by jbdemonte** [prisma/prisma#15831] [Q&A]
I'm looking to produce this query: 

`select distinct(adr.city) from client inner join client_address adr on client."billingAddressId" = adr.id`

I need to target it from client, but I far I saw till now, it does not seems to be possible to distinct city from client?

    prisma.client.findMany({
          // distinct: ['city'],       <= not possible 
          where: { companyId, deleted: false },
          select: {
            billingAddress: { 
                 // distinct: ['city'],       <= not possible
                 select: { city: true } 
            },
          },
        });
  
Am I wrong? is there a way to achieve it?

**Comment by nurul3101** [comment]
Hello @jbdemonte 👋 
Could you please share the schema file, especially the client and address models?
As a workaround, you can directly pass the query to [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) to execute it.

**Comment by jbdemonte** [comment]
Here is an excerpt of the schema:

```
model Client {
  id                      String            @id(map: "client_id") @db.VarChar
  name                    String            @db.VarChar

  billingAddressId        String            @unique @db.VarChar
  shippingAddressId       String?           @unique @db.VarChar

  billingAddress          ClientAddress     @relation("BillingAddress", fields: [billingAddressId], references: [id])
  shippingAddress         ClientAddress?    @relation("ShippingAddress", fields: [shippingAddressId], references: [id])

  @@map("client")
}

model ClientAddress {
  id                    String  @id(map: "client_address_id") @db.VarChar
  streetAddress         String  @db.VarChar
  additionalAddress     String  @db.VarChar
  postCode              String  @db.VarChar
  city                  String  @db.VarChar
  country               String  @db.VarChar
  billingAddressClient  Client? @relation("BillingAddress")
  shippingAddressClient Client? @relation("ShippingAddress")

  @@map("client_address")
}
```

I would prefer to avoid raw query for such a simple query

**Reply by nurul3101** [reply]
I can see that as city is not a part of Client model, due to which it isn't available while applying distinct clause. I would suggest to create a [Feature Request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that fields from relation models can also be supported in future.

**Reply by pantharshit00** [reply]
Hello @jbdemonte 

If you have already created the feature request, please leave a link to it here. That way other people landing here can follow it.

**Reply by piradata** [reply]
i think #24052 is basically that

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
